In [87]:
import numpy as np

In [88]:
import pandas as pd

In [89]:
import datetime as dt

In [90]:
import matplotlib.pyplot as plt

In [91]:
import seaborn as sns

In [92]:
import scipy.stats as stats

### Business Problem 1

In [93]:
loan_data=pd.read_csv("C:/Users/Rohit/Downloads/Python/Case Study/5 Hypothesis Testing/LoansData.csv")
loan_data

Amount.Requested  Amount.Funded.By.Investors Interest.Rate Loan.Length  \
0              20000.0                    20000.00         8.90%   36 months   
1              19200.0                    19200.00        12.12%   36 months   
2              35000.0                    35000.00        21.98%   60 months   
3              10000.0                     9975.00         9.99%   36 months   
4              12000.0                    12000.00        11.71%   36 months   
...                ...                         ...           ...         ...   
2495           30000.0                    29950.00        16.77%   60 months   
2496           16000.0                    16000.00        14.09%   60 months   
2497           10000.0                    10000.00        13.99%   36 months   
2498            6000.0                     6000.00        12.42%   36 months   
2499            9000.0                     5242.75        13.79%   36 months   

            Loan.Purpose Debt.To.Income.Ratio State Home.Ownership  \
0     debt_consolidation               14.90%    SC       MORTGAGE   
1     debt_consolidation               28.36%    TX       MORTGAGE   
2     debt_consolidation               23.81%    CA       MORTGAGE   
3     debt_consolidation               14.30%    KS       MORTGAGE   
4            credit_card               18.78%    NJ           RENT   
...                  ...                  ...   ...            ...   
2495  debt_consolidation               19.23%    NY       MORTGAGE   
2496    home_improvement               21.54%    MD            OWN   
2497  debt_consolidation                4.89%    PA       MORTGAGE   
2498      major_purchase               16.66%    NJ           RENT   
2499  debt_consolidation                6.76%    NY           RENT   

      Monthly.Income FICO.Range  Open.CREDIT.Lines  Revolving.CREDIT.Balance  \
0            6541.67    735-739               14.0                   14272.0   
1            4583.33    715-719               12.0                   11140.0   
2           11500.00    690-694               14.0                   21977.0   
3            3833.33    695-699               10.0                    9346.0   
4            3195.00    695-699               11.0                   14469.0   
...              ...        ...                ...                       ...   
2495         9250.00    705-709               15.0                   45880.0   
2496         8903.25    740-744               18.0                   18898.0   
2497         2166.67    680-684                4.0                    4544.0   
2498         3500.00    675-679                8.0                    7753.0   
2499         3875.00    670-674                7.0                    7589.0   

      Inquiries.in.the.Last.6.Months Employment.Length  
0                                2.0          < 1 year  
1                                1.0           2 years  
2                                1.0           2 years  
3                                0.0           5 years  
4                                0.0           9 years  
...                              ...               ...  
2495                             1.0           8 years  
2496                             1.0         10+ years  
2497                             0.0         10+ years  
2498                             0.0           5 years  
2499                             0.0         10+ years  

[2500 rows x 14 columns]

In [94]:
loan_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Amount.Requested                2499 non-null   float64
 1   Amount.Funded.By.Investors      2499 non-null   float64
 2   Interest.Rate                   2498 non-null   object 
 3   Loan.Length                     2500 non-null   object 
 4   Loan.Purpose                    2500 non-null   object 
 5   Debt.To.Income.Ratio            2499 non-null   object 
 6   State                           2500 non-null   object 
 7   Home.Ownership                  2499 non-null   object 
 8   Monthly.Income                  2499 non-null   float64
 9   FICO.Range                      2498 non-null   object 
 10  Open.CREDIT.Lines               2497 non-null   float64
 11  Revolving.CREDIT.Balance        2497 non-null   float64
 12  Inquiries.in.the.Last.6.Months  24

In [95]:
loan_data.shape

(2500, 14)

In [96]:
loan_data.isna().sum()

Amount.Requested                   1
Amount.Funded.By.Investors         1
Interest.Rate                      2
Loan.Length                        0
Loan.Purpose                       0
Debt.To.Income.Ratio               1
State                              0
Home.Ownership                     1
Monthly.Income                     1
FICO.Range                         2
Open.CREDIT.Lines                  3
Revolving.CREDIT.Balance           3
Inquiries.in.the.Last.6.Months     3
Employment.Length                 77
dtype: int64

In [97]:
loan_data.columns

Index(['Amount.Requested', 'Amount.Funded.By.Investors', 'Interest.Rate',
       'Loan.Length', 'Loan.Purpose', 'Debt.To.Income.Ratio', 'State',
       'Home.Ownership', 'Monthly.Income', 'FICO.Range', 'Open.CREDIT.Lines',
       'Revolving.CREDIT.Balance', 'Inquiries.in.the.Last.6.Months',
       'Employment.Length'],
      dtype='object')

In [98]:
loan_data.columns=loan_data.columns.str.replace('.', '_', regex=True)

In [99]:
loan_data.head(3)

Amount_Requested  Amount_Funded_By_Investors Interest_Rate Loan_Length  \
0           20000.0                     20000.0         8.90%   36 months   
1           19200.0                     19200.0        12.12%   36 months   
2           35000.0                     35000.0        21.98%   60 months   

         Loan_Purpose Debt_To_Income_Ratio State Home_Ownership  \
0  debt_consolidation               14.90%    SC       MORTGAGE   
1  debt_consolidation               28.36%    TX       MORTGAGE   
2  debt_consolidation               23.81%    CA       MORTGAGE   

   Monthly_Income FICO_Range  Open_CREDIT_Lines  Revolving_CREDIT_Balance  \
0         6541.67    735-739               14.0                   14272.0   
1         4583.33    715-719               12.0                   11140.0   
2        11500.00    690-694               14.0                   21977.0   

   Inquiries_in_the_Last_6_Months Employment_Length  
0                             2.0          < 1 year  
1                             1.0           2 years  
2                             1.0           2 years

In [100]:
loan_data['Interest_Rate']=loan_data['Interest_Rate'].str.replace('%', '').astype('float64')

In [101]:
loan_data['Debt_To_Income_Ratio']=loan_data['Debt_To_Income_Ratio'].str.replace('%', '').astype('float64')

In [102]:
loan_data

Amount_Requested  Amount_Funded_By_Investors  Interest_Rate Loan_Length  \
0              20000.0                    20000.00           8.90   36 months   
1              19200.0                    19200.00          12.12   36 months   
2              35000.0                    35000.00          21.98   60 months   
3              10000.0                     9975.00           9.99   36 months   
4              12000.0                    12000.00          11.71   36 months   
...                ...                         ...            ...         ...   
2495           30000.0                    29950.00          16.77   60 months   
2496           16000.0                    16000.00          14.09   60 months   
2497           10000.0                    10000.00          13.99   36 months   
2498            6000.0                     6000.00          12.42   36 months   
2499            9000.0                     5242.75          13.79   36 months   

            Loan_Purpose  Debt_To_Income_Ratio State Home_Ownership  \
0     debt_consolidation                 14.90    SC       MORTGAGE   
1     debt_consolidation                 28.36    TX       MORTGAGE   
2     debt_consolidation                 23.81    CA       MORTGAGE   
3     debt_consolidation                 14.30    KS       MORTGAGE   
4            credit_card                 18.78    NJ           RENT   
...                  ...                   ...   ...            ...   
2495  debt_consolidation                 19.23    NY       MORTGAGE   
2496    home_improvement                 21.54    MD            OWN   
2497  debt_consolidation                  4.89    PA       MORTGAGE   
2498      major_purchase                 16.66    NJ           RENT   
2499  debt_consolidation                  6.76    NY           RENT   

      Monthly_Income FICO_Range  Open_CREDIT_Lines  Revolving_CREDIT_Balance  \
0            6541.67    735-739               14.0                   14272.0   
1            4583.33    715-719               12.0                   11140.0   
2           11500.00    690-694               14.0                   21977.0   
3            3833.33    695-699               10.0                    9346.0   
4            3195.00    695-699               11.0                   14469.0   
...              ...        ...                ...                       ...   
2495         9250.00    705-709               15.0                   45880.0   
2496         8903.25    740-744               18.0                   18898.0   
2497         2166.67    680-684                4.0                    4544.0   
2498         3500.00    675-679                8.0                    7753.0   
2499         3875.00    670-674                7.0                    7589.0   

      Inquiries_in_the_Last_6_Months Employment_Length  
0                                2.0          < 1 year  
1                                1.0           2 years  
2                                1.0           2 years  
3                                0.0           5 years  
4                                0.0           9 years  
...                              ...               ...  
2495                             1.0           8 years  
2496                             1.0         10+ years  
2497                             0.0         10+ years  
2498                             0.0           5 years  
2499                             0.0         10+ years  

[2500 rows x 14 columns]

In [103]:
loan_data_con=loan_data.select_dtypes(['int', 'float64']).columns

In [104]:
loan_data_cat=loan_data.select_dtypes('object').columns

In [105]:
for i in loan_data_con:
    loan_data[i].fillna(loan_data[i].mean(), inplace=True)

In [106]:
for i in loan_data_cat:
    loan_data[i].fillna(loan_data[i].mode()[0], inplace=True)

In [107]:
loan_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Amount_Requested                2500 non-null   float64
 1   Amount_Funded_By_Investors      2500 non-null   float64
 2   Interest_Rate                   2500 non-null   float64
 3   Loan_Length                     2500 non-null   object 
 4   Loan_Purpose                    2500 non-null   object 
 5   Debt_To_Income_Ratio            2500 non-null   float64
 6   State                           2500 non-null   object 
 7   Home_Ownership                  2500 non-null   object 
 8   Monthly_Income                  2500 non-null   float64
 9   FICO_Range                      2500 non-null   object 
 10  Open_CREDIT_Lines               2500 non-null   float64
 11  Revolving_CREDIT_Balance        2500 non-null   float64
 12  Inquiries_in_the_Last_6_Months  25

##### Q a. Intrest rate is varied for different loan amounts (Less intrest charged for high loan amounts)

**i. H0:**
           Interest rate is not varying for different loan amount 
           
   **ii. Ha:**
           Interest rate is  varying for different loan amount
   
   **iii. CI = 95%**
      pvalue = 0.05
       
   **iv. Perform pearson correlation:**


In [108]:
int_rate=loan_data['Interest_Rate']

In [109]:
amt_req=loan_data['Amount_Requested']

In [110]:
stats.pearsonr(loan_data.Interest_Rate, loan_data.Amount_Requested)

(0.3324464059665682, 1.4148683739824417e-65)

v. Business Conclusion: 
* We reject the null hypopthesis
            there is a direct near to weak relation between interest rate and loan amount

#### Qb Loan length is directly effecting interest rate?

In [111]:
len_36mon_interest = loan_data.loc[loan_data.Loan_Length=='36 months', 'Interest_Rate']
len_36mon_interest

0        8.90
1       12.12
3        9.99
4       11.71
5       15.31
        ...  
2492    10.08
2494    14.65
2497    13.99
2498    12.42
2499    13.79
Name: Interest_Rate, Length: 1952, dtype: float64

In [112]:
len_60mon_interest = loan_data.loc[loan_data.Loan_Length=='60 months', 'Interest_Rate']
len_60mon_interest

2       21.98
7       17.14
12      21.67
15      15.65
17      10.37
        ...  
2472    22.45
2479    15.80
2493    23.28
2495    16.77
2496    14.09
Name: Interest_Rate, Length: 548, dtype: float64

**i. H0:**
           Loan length is not effecting interest rate 
           
   **ii. Ha:**
           Loan length is effecting interest rate
   
   **iii. CI = 95%**
      pvalue = 0.05
       
   **iv. Perform ttest:**

In [113]:
stats.ttest_ind(len_36mon_interest, len_60mon_interest)

Ttest_indResult(statistic=-23.41049023613487, pvalue=9.549596064332559e-110)

v. Business Conclusion: 
* We reject the null hypopthesis
            Loan length is effeceting the interest rate

#### Qc Interest rate varies for different purpose of loans?

In [114]:
loan_data.Loan_Purpose.value_counts()

debt_consolidation    1307
credit_card            444
other                  201
home_improvement       152
major_purchase         101
small_business          87
car                     50
wedding                 39
medical                 30
moving                  29
vacation                21
house                   20
educational             15
renewable_energy         4
Name: Loan_Purpose, dtype: int64

In [115]:
int_rate_debt_consolidation = loan_data.loc[ loan_data.Loan_Purpose == 'debt_consolidation', 'Interest_Rate']
int_rate_debt_consolidation

0        8.90
1       12.12
2       21.98
3        9.99
6        7.90
        ...  
2492    10.08
2494    14.65
2495    16.77
2497    13.99
2499    13.79
Name: Interest_Rate, Length: 1307, dtype: float64

In [116]:
int_rate_credit_card = loan_data.loc[ loan_data.Loan_Purpose == 'credit_card', 'Interest_Rate']
int_rate_credit_card

4       11.710000
7       17.140000
8       14.330000
9        6.910000
18       9.760000
          ...    
2463     7.900000
2467     7.290000
2476    14.090000
2478    11.710000
2484    13.065388
Name: Interest_Rate, Length: 444, dtype: float64

In [117]:
int_rate_other = loan_data.loc[ loan_data.Loan_Purpose == 'other', 'Interest_Rate']
int_rate_other

5       15.31
28       8.59
48       7.49
51      17.44
57       9.99
        ...  
2443    14.59
2450    14.79
2468     7.29
2486    17.27
2493    23.28
Name: Interest_Rate, Length: 201, dtype: float64

In [118]:
int_rate_home_improvement = loan_data.loc[ loan_data.Loan_Purpose == 'home_improvement', 'Interest_Rate']
int_rate_home_improvement

37       7.62
39      12.49
40      17.27
69      14.09
88       7.51
        ...  
2461     7.90
2462    11.49
2465     6.03
2466    22.47
2496    14.09
Name: Interest_Rate, Length: 152, dtype: float64

In [119]:
int_rate_major_purchase = loan_data.loc[ loan_data.Loan_Purpose == 'major_purchase', 'Interest_Rate']
int_rate_major_purchase

45      11.86
95       6.62
152     10.75
163     10.16
175     10.38
        ...  
2421     5.79
2432    11.49
2473    15.13
2475    14.09
2498    12.42
Name: Interest_Rate, Length: 101, dtype: float64

In [120]:
int_rate_small_business = loan_data.loc[ loan_data.Loan_Purpose == 'small_business', 'Interest_Rate']
int_rate_small_business

92      15.81
104     10.65
110      7.90
117     14.09
136      6.62
        ...  
2435    20.52
2454    18.64
2480     6.03
2482     7.51
2483    14.33
Name: Interest_Rate, Length: 87, dtype: float64

In [121]:
int_rate_car = loan_data.loc[ loan_data.Loan_Purpose == 'car', 'Interest_Rate']
int_rate_car

22      17.99
33      15.80
35       7.90
86       8.90
176      7.90
292      6.03
346     17.04
370     14.09
389     14.09
394     19.72
418      6.91
539      7.49
574     10.59
590      6.99
638     11.83
696     13.43
789     11.71
891     13.67
904     13.99
941     13.47
960     11.71
984     11.12
1005     7.88
1016    10.00
1059    13.16
1079    13.23
1149    17.99
1192     6.62
1215     6.03
1325     6.03
1379     5.79
1410     8.90
1424     8.90
1496     9.25
1627     7.90
1633     7.62
1658     7.62
1707    17.14
1880     9.63
1921    13.11
1950     6.62
1985     7.49
1991    13.49
2011    10.36
2116    17.49
2209    18.25
2338     7.90
2402    13.93
2434    14.33
2444     6.62
Name: Interest_Rate, dtype: float64

In [122]:
int_rate_wedding = loan_data.loc[ loan_data.Loan_Purpose == 'wedding', 'Interest_Rate']
int_rate_wedding

78      17.770000
273     15.270000
341      7.900000
361     16.490000
408      6.620000
429      9.910000
432      7.900000
453      7.620000
565     14.090000
642     10.370000
678      7.900000
693     13.720000
770     12.690000
808     10.250000
811     18.490000
821     12.980000
885      7.900000
931      9.990000
1064    13.110000
1288     7.620000
1291     8.000000
1313     7.900000
1319     9.910000
1345    12.120000
1370     5.420000
1482    14.420000
1483    12.840000
1705    15.310000
1904    12.120000
1964    17.880000
1965    12.420000
2035    11.860000
2124    13.110000
2206    15.310000
2222     9.630000
2272    16.290000
2366    12.210000
2472    22.450000
2487    13.065388
Name: Interest_Rate, dtype: float64

In [123]:
int_rate_medical = loan_data.loc[ loan_data.Loan_Purpose == 'medical', 'Interest_Rate']
int_rate_medical

65       7.90
234     14.09
261     14.65
291     15.80
294     14.12
340      5.42
423      7.29
837     23.28
848      7.90
879     13.11
1250     6.03
1293     6.62
1309     7.62
1355     6.03
1576    13.11
1577     7.90
1582    16.29
1690    14.27
1779    13.57
1852     6.99
1854    14.22
1946    17.49
2000    10.59
2022     7.49
2038    19.69
2123     7.51
2182     6.54
2189    13.98
2238    13.99
2451    16.00
Name: Interest_Rate, dtype: float64

In [124]:
int_rate_moving = loan_data.loc[ loan_data.Loan_Purpose == 'moving', 'Interest_Rate']
int_rate_moving

10      19.72
195      6.17
209     13.11
228     24.89
349     10.99
435      5.79
497     12.21
524     14.84
659     14.33
756     15.31
1046    11.71
1152     6.76
1183    11.83
1347    12.68
1420    10.75
1711    19.05
1729     7.88
1870    10.38
1929    23.76
1960    14.65
1974    17.49
2082    15.31
2106    14.83
2167    12.42
2251    14.22
2374     7.51
2383    19.72
2415    15.96
2469    10.74
Name: Interest_Rate, dtype: float64

In [125]:
int_rate_vacation = loan_data.loc[ loan_data.Loan_Purpose == 'vacation', 'Interest_Rate']
int_rate_vacation

24      16.82
111      6.03
130      9.76
575      7.29
1006    12.12
1134    16.77
1141    13.98
1197     7.88
1200    18.39
1302    14.59
1308     5.42
1349    13.67
1688     8.94
1802    10.99
1806    11.49
1850    14.65
1895    15.80
1952     6.91
2033    17.56
2054     7.90
2367    14.33
Name: Interest_Rate, dtype: float64

In [126]:
int_rate_house = loan_data.loc[ loan_data.Loan_Purpose == 'house', 'Interest_Rate']
int_rate_house

43      21.74
339     22.47
369     13.11
536     15.21
806      9.63
936      7.90
1035    14.96
1067     6.03
1114    15.96
1387    10.37
1389    21.67
1573    16.32
1607     6.62
1685    11.83
1823    16.82
2036    13.06
2197    13.06
2225     5.99
2387    12.12
2403    14.09
Name: Interest_Rate, dtype: float64

In [127]:
int_rate_educational = loan_data.loc[ loan_data.Loan_Purpose == 'educational', 'Interest_Rate']
int_rate_educational

59       8.59
300     18.67
401     12.29
940     12.61
974     13.48
985      7.75
1137    14.61
1145     7.88
1348     6.76
1649     9.88
1686    13.75
1888    12.87
2028     7.40
2264     9.63
2310     8.94
Name: Interest_Rate, dtype: float64

In [128]:
int_rate_renewable_energy = loan_data.loc[ loan_data.Loan_Purpose == 'renewable_energy', 'Interest_Rate']
int_rate_renewable_energy

160      6.92
289     10.36
1060     7.90
2246    14.33
Name: Interest_Rate, dtype: float64

**i. H0:**
           Interest rate doesn't vary for different purpose of loan
                         
   **ii. Ha:**
           Interest rate vary for different purpose of loan
                           
   **iii. CI = 95%**
      pvalue = 0.05
       
   **iv. Perform f test:**

In [129]:
stats.f_oneway(int_rate_debt_consolidation, int_rate_credit_card, int_rate_other, int_rate_home_improvement, int_rate_major_purchase, int_rate_small_business, int_rate_car, int_rate_wedding, int_rate_medical, int_rate_moving, int_rate_vacation, int_rate_house, int_rate_educational, int_rate_renewable_energy)

F_onewayResult(statistic=7.4826387840974355, pvalue=1.1599665535989115e-14)

v. Business Conclusion: 
* We reject the null hypopthesis
            Interest rate vary for different purpose of loan

In [130]:
loan_data.FICO_Range.nunique()

38

#### Qd. There is relationship between FICO scores and Home Ownership. It means that, People with owning home will have high FICO scores.

**i. H0:**
           There is no relationship between FICO scores and home ownership
           
   **ii. Ha:**
          There is a relationship between FICO scores and home ownership
   
   **iii. CI = 95%**
      pvalue = 0.05
       
   **iv. Perform Chi-square test:**

In [131]:
loan_data.head(3)

Amount_Requested  Amount_Funded_By_Investors  Interest_Rate Loan_Length  \
0           20000.0                     20000.0           8.90   36 months   
1           19200.0                     19200.0          12.12   36 months   
2           35000.0                     35000.0          21.98   60 months   

         Loan_Purpose  Debt_To_Income_Ratio State Home_Ownership  \
0  debt_consolidation                 14.90    SC       MORTGAGE   
1  debt_consolidation                 28.36    TX       MORTGAGE   
2  debt_consolidation                 23.81    CA       MORTGAGE   

   Monthly_Income FICO_Range  Open_CREDIT_Lines  Revolving_CREDIT_Balance  \
0         6541.67    735-739               14.0                   14272.0   
1         4583.33    715-719               12.0                   11140.0   
2        11500.00    690-694               14.0                   21977.0   

   Inquiries_in_the_Last_6_Months Employment_Length  
0                             2.0          < 1 year  
1                             1.0           2 years  
2                             1.0           2 years

In [132]:
observ_table=pd.crosstab(index=loan_data.FICO_Range, columns=loan_data.Home_Ownership)
observ_table

Home_Ownership  MORTGAGE  NONE  OTHER  OWN  RENT
FICO_Range                                      
640-644                0     0      0    1     4
645-649                1     0      0    0     2
650-654                1     0      0    0     0
655-659                0     0      1    0     3
660-664               41     0      0   17    67
665-669               52     0      0   18    75
670-674               68     0      1   13    91
675-679               79     0      1   11    75
680-684               55     0      2    9    91
685-689               61     0      0   10    65
690-694               55     0      0   10    75
695-699               68     0      0   10    75
700-704               60     0      0   12    59
705-709               66     0      0   12    56
710-714               55     0      0   10    47
715-719               46     0      0    2    45
720-724               49     0      0    1    64
725-729               48     0      0    5    41
730-734               46     0      0    7    41
735-739               30     0      0    6    29
740-744               26     0      0    4    23
745-749               26     0      0    7    21
750-754               32     0      0    4    25
755-759               28     0      0    6    12
760-764               28     0      0    4    14
765-769               22     0      0    3    11
770-774               12     0      0    1     4
775-779               13     0      0    3     6
780-784               21     0      0    3     4
785-789               12     0      0    2     5
790-794               15     0      0    1     4
795-799                7     0      0    1     5
800-804                7     1      0    3     2
805-809                6     0      0    2     3
810-814                5     0      0    1     2
815-819                6     0      0    0     0
820-824                0     0      0    1     0
830-834                1     0      0    0     0

In [133]:
stats.chi2_contingency(observ_table)

(473.564802816852,
 1.0065160858472195e-35,
 148,
 array([[2.29600e+00, 2.00000e-03, 1.00000e-02, 4.00000e-01, 2.29200e+00],
        [1.37760e+00, 1.20000e-03, 6.00000e-03, 2.40000e-01, 1.37520e+00],
        [4.59200e-01, 4.00000e-04, 2.00000e-03, 8.00000e-02, 4.58400e-01],
        [1.83680e+00, 1.60000e-03, 8.00000e-03, 3.20000e-01, 1.83360e+00],
        [5.74000e+01, 5.00000e-02, 2.50000e-01, 1.00000e+01, 5.73000e+01],
        [6.65840e+01, 5.80000e-02, 2.90000e-01, 1.16000e+01, 6.64680e+01],
        [7.94416e+01, 6.92000e-02, 3.46000e-01, 1.38400e+01, 7.93032e+01],
        [7.62272e+01, 6.64000e-02, 3.32000e-01, 1.32800e+01, 7.60944e+01],
        [7.20944e+01, 6.28000e-02, 3.14000e-01, 1.25600e+01, 7.19688e+01],
        [6.24512e+01, 5.44000e-02, 2.72000e-01, 1.08800e+01, 6.23424e+01],
        [6.42880e+01, 5.60000e-02, 2.80000e-01, 1.12000e+01, 6.41760e+01],
        [7.02576e+01, 6.12000e-02, 3.06000e-01, 1.22400e+01, 7.01352e+01],
        [6.01552e+01, 5.24000e-02, 2.62000e-01, 1.

v. Business Conclusion: 
* We reject the null hypothesis T
             There is a relationship between FICO scores and null hypothesis

### Business Problem 2

In [134]:
price_quote_data=pd.read_csv("C:/Users/Rohit/Downloads/Python/Case Study/5 Hypothesis Testing/Price_Quotes.csv")
price_quote_data

Order_Number  Barry_Price  Mary_Price
0              1          126         114
1              2          110         118
2              3          138         114
3              4          142         111
4              5          146         129
5              6          136         119
6              7           94          97
7              8          103         104
8              9          140         127
9             10          152         133
10            11          108         103
11            12           97         108

In [135]:
price_quote_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype
---  ------        --------------  -----
 0   Order_Number  12 non-null     int64
 1   Barry_Price   12 non-null     int64
 2   Mary_Price    12 non-null     int64
dtypes: int64(3)
memory usage: 416.0 bytes


In [136]:
price_quote_data.Barry_Price.mean()

124.33333333333333

In [137]:
price_quote_data.Mary_Price.mean()

114.75

**i. H0:**
           There is no difference in the average price quote by Barry and Mary
           
   **ii. Ha:**
              There is a diference in the average price quote by Barry and Mary
   
   **iii. CI = 95%**
      pvalue = 0.05
       
   **iv. Perform ttest :**

In [138]:
stats.ttest_ind(price_quote_data.Barry_Price, price_quote_data.Mary_Price)

Ttest_indResult(statistic=1.4147436739281787, pvalue=0.17114226132118285)

v. Business Conclusion: 
* We fail to reject the null hypopthesis
           we are unable to find diference in the average price quote by Barry and Mary

#### Determine what effect, if any, the reengineering effort had on the incidence behavioral problems and staff turnover. i.e To determine if the reengineering effort changed the critical incidence rate. Isthere evidence that the critical incidence rate improved?

### Business Problem 3|

In [139]:
treat_facility=pd.read_csv("C:/Users/Rohit/Downloads/Python/Case Study/5 Hypothesis Testing/Treatment_Facility.csv")
treat_facility

Month Reengineer  Employee_Turnover       VAR4        VAR5
0       1      Prior             0.0000  24.390244   42.682927
1       2      Prior             6.0606  19.354839   25.806452
2       3      Prior            12.1212  35.087719  146.198830
3       4      Prior             3.3333  18.404908  110.429448
4       5      Prior            12.9032  17.964072   23.952096
5       6      Prior             9.6774  41.176471   47.058824
6       7      Prior            11.7647  13.422819    0.000000
7       8      Prior            11.4286  31.250000   25.000000
8       9      Prior            23.0769  17.241379  132.183908
9      10      Prior            15.0000  16.574586   16.574586
10     11      Prior            10.0000   5.917160   88.757396
11     12      Prior            26.0870  21.164021   31.746032
12     13      Prior            11.1111   5.076142   10.152284
13     14       Post            26.6667   9.708738   29.126214
14     15       Post            17.9487   4.201681   21.008403
15     16       Post            12.8205  13.043478   34.782609
16     17       Post            15.7895   9.950249   19.900498
17     18       Post             0.0000   9.950249    9.950249
18     19       Post            30.7692   8.510638   25.531915
19     20       Post            26.8293   9.259259   23.148148

In [140]:
treat_facility.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Month              20 non-null     int64  
 1   Reengineer         20 non-null     object 
 2   Employee_Turnover  20 non-null     float64
 3   VAR4               20 non-null     float64
 4   VAR5               20 non-null     float64
dtypes: float64(3), int64(1), object(1)
memory usage: 928.0+ bytes


In [141]:
treat_facility.rename(columns = {'VAR4' : 'TRFF', 'VAR5' : 'CI'}, inplace =True)

In [142]:
prior_ci = treat_facility.loc[treat_facility.Reengineer=='Prior', 'CI']
prior_ci

0      42.682927
1      25.806452
2     146.198830
3     110.429448
4      23.952096
5      47.058824
6       0.000000
7      25.000000
8     132.183908
9      16.574586
10     88.757396
11     31.746032
12     10.152284
Name: CI, dtype: float64

In [143]:
prior_ci_7months=prior_ci.loc[6:13:1, ]

In [144]:
prior_ci_7months.mean()

43.487743735

In [145]:
post_ci = treat_facility.loc[treat_facility.Reengineer=='Post', 'CI']
post_ci

13    29.126214
14    21.008403
15    34.782609
16    19.900498
17     9.950249
18    25.531915
19    23.148148
Name: CI, dtype: float64

In [146]:
post_ci.mean()

23.34971927988571

**i. H0:**
           u_prior < u_post 
           
   **ii. Ha:**
           u_prior > u_post
   
   **iii. CI = 95%**
      pvalue = 0.05
       
   **iv. Perform ttest:**


In [147]:
stats.ttest_rel(prior_ci_7months, post_ci)

Ttest_relResult(statistic=1.1053112945653105, pvalue=0.31136482510821195)

5. Business Conclusion:
    
    new p_value > p_values
    
    0.311 > 0.05
    
    accpet the H0!
    
    we can't say that reengineering effort improved critical incidence rate

### Business Problem 4

In [148]:
priority_asessment=pd.read_csv("C:/Users/Rohit/Downloads/Python/Case Study/5 Hypothesis Testing/Priority_Assessment.csv")
priority_asessment

Days Priority
0     3.3     High
1     7.9   Medium
2     0.3     High
3     0.7   Medium
4     8.6   Medium
..    ...      ...
637   2.5      Low
638   0.3     High
639   0.3   Medium
640   1.3   Medium
641   0.5     High

[642 rows x 2 columns]

In [149]:
priority_asessment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 642 entries, 0 to 641
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Days      642 non-null    float64
 1   Priority  642 non-null    object 
dtypes: float64(1), object(1)
memory usage: 10.2+ KB


In [150]:
high_priority=priority_asessment.loc[priority_asessment.Priority=='High', 'Days']

In [151]:
medium_priority=priority_asessment.loc[priority_asessment.Priority=='Medium', 'Days']

In [152]:
low_priority=priority_asessment.loc[priority_asessment.Priority=='Low', 'Days']

In [153]:
print('Mean of high priority:', high_priority.mean())
print('Mean of medium priority:', medium_priority.mean())
print('Mean of low priority:', low_priority.mean())

Mean of high priority: 3.023619631901845
Mean of medium priority: 2.5000000000000004
Mean of low priority: 4.228358208955225


**i. H0:**
           prioritization system doesn't speeds up completion of high priority task  
           
   **ii. Ha:**
          prioritization system speeds up completion of high priority task 
   
   **iii. CI = 95%**
      pvalue = 0.05
       
   **iv. Perform f-test:**

In [154]:
stats.f_oneway(high_priority, medium_priority, low_priority)

F_onewayResult(statistic=1.812311010076072, pvalue=0.16411459461716182)

v. Business conclusion:

    new p_value > p_value
    
    0.16 > 0.05
    
    accpet the H0!
    
    prioritization system brings no change in speeding completion of high priority task 

### Business Problem 5

In [155]:
films=pd.read_csv("C:/Users/Rohit/Downloads/Python/Case Study/5 Hypothesis Testing/Films.csv")
films

_rowstate_                    Movie  Gender Marital_Status  Sinage  \
0             0  Ferris Buellers Day Off  Female        Married     2.0   
1             0  Ferris Buellers Day Off  Female         Single     1.0   
2             0  Ferris Buellers Day Off    Male        Married     2.0   
3             0  Ferris Buellers Day Off  Female        Married     1.0   
4             0  Ferris Buellers Day Off  Female        Married     1.0   
..          ...                      ...     ...            ...     ...   
325           0               Old School       2              2     1.0   
326           0               Old School       1              1     2.0   
327           0               Old School       2              1     2.0   
328           0               Old School       1              1     1.0   
329           0               Old School       2              2     2.0   

     Parking  Clean  Overall  Age  Income Hear_About  
0        2.0    2.0      2.0  3.0     1.0          5  
1        1.0    1.0      1.0  2.0     1.0          5  
2        4.0    3.0      2.0  4.0     1.0          5  
3        3.0    2.0      2.0  4.0     1.0          5  
4        1.0    1.0      1.0  3.0     3.0          1  
..       ...    ...      ...  ...     ...        ...  
325      2.0    1.0      1.0  2.0     1.0          1  
326      2.0    2.0      2.0  3.0     3.0          5  
327      1.0    1.0      2.0  2.0     2.0          5  
328      1.0    1.0      1.0  2.0     1.0          3  
329      2.0    3.0      1.0  2.0     1.0          5  

[330 rows x 11 columns]

In [156]:
films.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 330 entries, 0 to 329
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   _rowstate_      330 non-null    int64  
 1   Movie           330 non-null    object 
 2   Gender          330 non-null    object 
 3   Marital_Status  328 non-null    object 
 4   Sinage          328 non-null    float64
 5   Parking         328 non-null    float64
 6   Clean           327 non-null    float64
 7   Overall         328 non-null    float64
 8   Age             328 non-null    float64
 9   Income          314 non-null    float64
 10  Hear_About      323 non-null    object 
dtypes: float64(6), int64(1), object(4)
memory usage: 28.5+ KB


In [157]:
cat_col=films.columns.difference(['Income'])

In [158]:
films[cat_col]=films[cat_col].apply(lambda x: x.fillna(x.mode()[0]))

In [159]:
films.Income.fillna(films.Income.mean(), inplace= True)

In [160]:
films.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 330 entries, 0 to 329
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   _rowstate_      330 non-null    int64  
 1   Movie           330 non-null    object 
 2   Gender          330 non-null    object 
 3   Marital_Status  330 non-null    object 
 4   Sinage          330 non-null    float64
 5   Parking         330 non-null    float64
 6   Clean           330 non-null    float64
 7   Overall         330 non-null    float64
 8   Age             330 non-null    float64
 9   Income          330 non-null    float64
 10  Hear_About      330 non-null    object 
dtypes: float64(6), int64(1), object(4)
memory usage: 28.5+ KB


In [161]:
films['Gender']=films.Gender.str.replace('Male', '1')
films['Gender']=films.Gender.str.replace('Female', '2')

In [162]:
films['Marital_Status']=films.Gender.str.replace('Married', '1')
films['Marital_Status']=films.Gender.str.replace('Single', '2')

##### What is the overall level of customer satisfaction?

In [163]:
films.Overall.mode()[0]

2.0

The overall rating of customer satisfaction is Good

##### What factors are linked to satisfaction?

In [164]:
films.corr()

_rowstate_    Sinage   Parking     Clean   Overall       Age  \
_rowstate_         NaN       NaN       NaN       NaN       NaN       NaN   
Sinage             NaN  1.000000  0.469180  0.349144  0.382103 -0.089461   
Parking            NaN  0.469180  1.000000  0.440771  0.514429 -0.023468   
Clean              NaN  0.349144  0.440771  1.000000  0.350822 -0.000114   
Overall            NaN  0.382103  0.514429  0.350822  1.000000 -0.032904   
Age                NaN -0.089461 -0.023468 -0.000114 -0.032904  1.000000   
Income             NaN -0.036968 -0.023575  0.016643 -0.005568  0.136119   

              Income  
_rowstate_       NaN  
Sinage     -0.036968  
Parking    -0.023575  
Clean       0.016643  
Overall    -0.005568  
Age         0.136119  
Income      1.000000

Factors that are directly affecting satisfaction are  parking> signage > clean

##### What is the demographic profile of Film on the Rocks patrons?

In [165]:
print(films.Gender.value_counts())
print('More no. of females went for the film as compared to males')

2    213
1    117
Name: Gender, dtype: int64
More no. of females went for the film as compared to males


In [166]:
print(films.Age.value_counts())
print('Maximum people went for the films were of  Age group 13-30 and 31-60')

2.0    177
3.0    117
1.0     26
4.0     10
Name: Age, dtype: int64
Maximum people went for the films were of  Age group 13-30 and 31-60


In [167]:
print(films.Income.value_counts())
print('Maximum people went for the films were having income less than $50000')

1.000000    142
3.000000     90
2.000000     82
1.834395     16
Name: Income, dtype: int64
Maximum people went for the films were having income less than $50000


In [168]:
print(films.Marital_Status.value_counts())
print('Maximum people went for the film were Single')

2    213
1    117
Name: Marital_Status, dtype: int64
Maximum people went for the film were Single


##### In what media outlet(s) should the film series be advertised?

In [169]:
print(films.Hear_About.value_counts())
print('Fims should be advertised through word of mouth and on website')

5      233
4       41
1       22
3       14
2       12
2,5      2
3,4      2
4,5      1
1,5      1
5,4      1
3,5      1
Name: Hear_About, dtype: int64
Fims should be advertised through word of mouth and on website
